src/_pytest/assertion/util.py

```diff
@@ -169,23 +190,56 @@ def assertrepr_compare(config, op: str, left: Any, right: Any) -> Optional[List[
 
 
 def _compare_eq_any(left: Any, right: Any, verbose: int = 0) -> List[str]:
-    explanation = []  # type: List[str]
+    """
+    Compare 2 objects.
+    The generic comparison function, which calls other functions based on the input's type.
+
+    :param left
+    :type left: Any
+    :param right
+    :type right: Any
+    :param verbose
+    :type verbose: int
+    .. note:: indendation used for better visual perception of the comparison report.
+    :return: explanation of an AssertionError
+    :rtype: List[str]
+    """
+    explanation = []
+
+    my_name = cast(types.FrameType, inspect.currentframe()).f_code.co_name
+    num_calls = _get_number_of_calls(func_name=my_name)  # type: int
+    indentation = " " * 4 * (num_calls - 1)  # type: str
+
     if istext(left) and istext(right):
-        explanation = _diff_text(left, right, verbose)
+        explanation = _diff_text(left=left, right=right, verbose=verbose)
```

Comment:
Let's keep the arguments positional, here and below:

1. `left=left` is a bit redundant and longer, the positionals are a bit more readable here.
2. It is harder to review a diff when style changes are included, so it's better to submit them in a separate PR.